pkg/controller/storagecluster/reconcile.go

```diff
@@ -196,52 +186,54 @@ func (r *ReconcileStorageCluster) Reconcile(request reconcile.Request) (reconcil
-	scinit := &ocsv1.StorageClusterInitialization{}
-	err = r.client.Get(context.TODO(), request.NamespacedName, scinit)
-	if err != nil {
-		if errors.IsNotFound(err) {
-			reqLogger.Info("Creating StorageClusterInitialization resource")
-
-			// if the StorageClusterInitialization object doesn't exist
-			// ensure we re-reconcile on all initialization resources
-			instance.Status.StorageClassesCreated = false
-			instance.Status.CephObjectStoresCreated = false
-			instance.Status.CephBlockPoolsCreated = false
-			instance.Status.CephObjectStoreUsersCreated = false
-			instance.Status.CephFilesystemsCreated = false
-			instance.Status.FailureDomain = determineFailureDomain(instance)
-			err = r.client.Status().Update(context.TODO(), instance)
-			if err != nil {
-				return reconcile.Result{}, err
-			}
+		// Check for StorageClusterInitialization
+		scinit := &ocsv1.StorageClusterInitialization{}
+		err = r.client.Get(context.TODO(), request.NamespacedName, scinit)
+		if err != nil {
+			if errors.IsNotFound(err) {
+				reqLogger.Info("Creating StorageClusterInitialization resource")
+
+				// if the StorageClusterInitialization object doesn't exist
+				// ensure we re-reconcile on all initialization resources
+				instance.Status.StorageClassesCreated = false
+				instance.Status.CephObjectStoresCreated = false
+				instance.Status.CephBlockPoolsCreated = false
+				instance.Status.CephObjectStoreUsersCreated = false
+				instance.Status.CephFilesystemsCreated = false
+				instance.Status.FailureDomain = determineFailureDomain(instance)
+				err = r.client.Status().Update(context.TODO(), instance)
+				if err != nil {
+					return reconcile.Result{}, err
+				}
 
-			scinit.Name = request.Name
-			scinit.Namespace = request.Namespace
-			if err = controllerutil.SetControllerReference(instance, scinit, r.scheme); err != nil {
-				return reconcile.Result{}, err
-			}
+				scinit.Name = request.Name
+				scinit.Namespace = request.Namespace
+				if err = controllerutil.SetControllerReference(instance, scinit, r.scheme); err != nil {
+					return reconcile.Result{}, err
+				}
 
-			err = r.client.Create(context.TODO(), scinit)
-			switch {
-			case err == nil:
-				log.Info("Created StorageClusterInitialization resource")
-			case errors.IsAlreadyExists(err):
-				log.Info("StorageClusterInitialization resource already exists")
-			default:
-				log.Error(err, "Failed to create StorageClusterInitialization resource")
+				err = r.client.Create(context.TODO(), scinit)
+				switch {
+				case err == nil:
+					log.Info("Created StorageClusterInitialization resource")
+				case errors.IsAlreadyExists(err):
+					log.Info("StorageClusterInitialization resource already exists")
+				default:
+					log.Error(err, "Failed to create StorageClusterInitialization resource")
+					return reconcile.Result{}, err
+				}
+			} else {
+				// Error reading the object - requeue the request.
 				return reconcile.Result{}, err
 			}
```

Comment:
Let's move this out to a separate function (`ensureStorageClusterInitialization`) for sake of readability.

Reply:
Moved it to a separate function as suggested (`ensureStorageClusterInit`)...